CRM/CiviAwards/BAO/AwardDetail.php

```diff
@@ -76,4 +77,67 @@ private static function validateDates(array $params) {
     }
   }
 
+
+  /**
+   * Validates review_field options to ensure legal options are passed.
+   *
+   * @param array $params
+   *  Request Parameters
+   *
+   * @throws \Exception
+   */
+  private static function validateReviewFields(array &$params) {
+    $reviewFieldsLegalOptions = [
+      'id' => FILTER_VALIDATE_INT,
+      'weight' => FILTER_VALIDATE_INT,
+      'required' => FILTER_VALIDATE_BOOLEAN
+    ];
+    $reviewFields = CRM_Utils_Array::value('review_fields', $params);
+    if (empty($reviewFields)) {
+      return;
+    }
+
+    $filteredReviewFields = [];
+    foreach ($reviewFields as $reviewFieldOptions) {
+      $invalidReviewFieldOptions = [];
+      $validReviewFieldOptions = array_filter(
+        $reviewFieldOptions,
+        function($value, $key) use ($reviewFieldsLegalOptions, &$invalidReviewFieldOptions) {
```

Comment:
Can we move this to a separate function rather and use he function name in the array_filter here. Makes the code look less cluttered.